app/models/report.rb

```diff
@@ -14,4 +28,25 @@ def editable?(target_user)
   def created_on
     created_at.to_date
   end
+
+  def extract_mentions
+    content.scan(REPORT_URL_PATTERN).flatten.uniq
+  end
+
+  def update_mentions!
+    mention_relationships.destroy_all
+    mentions = extract_mentions
+    mentions.each do |mention_id|
+      Mention.create!(mention_id: id, mentioned_id: mention_id)
+    end
+  end
+
+  private
+
+  def no_self_mention
+    mentions = extract_mentions
+    return unless mentions.include?(id.to_s)
```

Comment:
ActiveSupportにある `exclude?` が使えそうです。
https://api.rubyonrails.org/classes/Enumerable.html#method-i-exclude-3F

ActiveSupportではRubyの標準メソッドからさらにメソッドを拡張して便利にしてあるので、一度ざっと見ておくとよいと思います。

```suggestion
    return if extract_mentions.exclude?(id.to_s)
```

Reply:
ありがとうございます。対義語にあたるメソッドは存在しそうですね。かなり数が多そうですが、ざっと目を通してみます！